File: HonestCoach.cpp
#include <bits/stdc++.h>
using namespace std;

int main(void) {
    // variable declaration and input
    int t;
    cin >> t;
    vector<int> res;
    while (t--) {
        int n;
        cin >> n;
        vector<int> arr(n);
        for (auto &x : arr)
            cin >> x;
        sort(arr.begin(), arr.end());

        int result = arr[n-1] - arr[0];
        for (int i=0; i<n; i++) {
            for (int j=i+1; j<n; j++) 
                result = min(result, arr[j]-arr[i]);
        }
        res.push_back(result);
    }
    
    // output and returning 0
    for (auto &x : res)
        cout << x << endl;

    return 0;
}

File: Python/HonestCoach.py
def main():
    t = int(input())
    res = []
    while t:
        n = int(input())
        arr = [int(input()) for i in range(n)]
        arr = sorted(arr)
        result = arr[n-1] - arr[0]
        for i in range(n):
            for j in range(i+1, n):
                result = min(result, arr[j]-arr[i])
        res.append(result)
        t -= 1
    for i in res:
        print(i)


if __name__ == "__main__":
    main()
